dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -630,4 +636,13 @@ private long getCurrentTime() {
 
     return keyFields;
   }
+
+  private List<String> getRulebookRulesFromTable(StructuredRow rulebookRow) {
+    String rulesColumn = Objects.toString(rulebookRow.getString(RULES_COL), "");
+    List<String> rules = new ArrayList<>(Arrays.asList(rulesColumn.split(",")));
```

Comment:
It doesn't seem right that empty rules are written in the first place. They should be removed immediately where they are added, instead of stored and then filtered out on reading.